http2_general/test_h2_headers.py

```diff
@@ -439,6 +440,125 @@ def __test(self, request: list):
         self.assertTrue(client.wait_for_connection_close())
 
 
+class TestIncorrectIfModifiedSince(H2Base):
+    tempesta = {
+        "config": """
+listen 80;
+listen 443 proto=h2;
+
+server ${server_ip}:8000;
+
+tls_certificate ${tempesta_workdir}/tempesta.crt;
+tls_certificate_key ${tempesta_workdir}/tempesta.key;
+tls_match_any_server_name;
+
+cache 2;
+cache_fulfill * *;
+cache_methods GET HEAD POST;
+""",
+    }
+
+    def __test(self, if_modified_since, huffman):
+        self.start_all_services()
+        client = self.get_client("deproxy")
+        server = self.get_server("deproxy")
+
+        server.set_response(
+            "HTTP/1.1 200 OK\r\n"
+            + "Content-Length: 0\r\n"
+            + "Server: Deproxy Server\r\n"
+            + "Date: Mon, 12 Dec 2016 13:59:39 GMT\r\n"
+            + "\r\n"
+        )
+
+        client.make_request(
+            client.create_request(method="GET", headers=[]),
+            huffman=huffman,
+        )
+        client.wait_for_response()
+        self.assertEqual(client.last_response.status, "200")
+
+        client.make_request(
+            client.create_request(method="GET", headers=[("if-modified-since", if_modified_since)]),
+            huffman=huffman,
+        )
+        client.wait_for_response()
+        self.assertEqual(client.last_response.status, "200")
+
+    @marks.Parameterize.expand(
+        [
+            marks.Param(
+                name="if_modified_since_huffman_1",
+                if_modified_since="Inv Jan 01 00:00:01 1970  Inv Jan 01 00:00:01 2222",
+                huffman=True,
+            ),
+            marks.Param(
+                name="if_modified_since_no_huffman_1",
+                if_modified_since="Inv Jan 01 00:00:01 1970  Inv Jan 01 00:00:01 2222",
+                huffman=False,
+            ),
+        ]
+    )
+    def test(self, name, if_modified_since, huffman):
+        self.__test(if_modified_since, huffman)
+
+    @marks.Parameterize.expand(
+        [
+            marks.Param(
+                name="if_modified_since_huffman_2",
+                if_modified_since="Sat, 29 Oct 2222 19:43:31 GMT111",
+                huffman=True,
+            ),
+            marks.Param(
+                name="if_modified_since_no_huffman_2",
+                if_modified_since="Sat, 29 Oct 2222 19:43:31 GMT111",
+                huffman=False,
+            ),
+        ]
+    )
+    @unittest.expectedFailure
+    def test(self, name, if_modified_since, huffman):
+        """
+        Tempesta FW doesn't check that there is any invalid
+        bytes after GMT in the date. 
+        """
+        self.__test(if_modified_since, huffman)
+
+    def test_many_if_modified_since(self):
```

Comment:
Please add description to the case. This is not obvious that we had a bug because of that `if-none-match` allowed to use second `if-modified-since`